### ISSUE TITLE
IOU - Error page appears on flash when navigate to IOU via link

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#46529]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.14-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4784667&group_by=cases:section_id&group_order=asc&group_id=229066
**Email or phone of affected tester (no customers):** gocemate+a768@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. As account A, create an IOU with account B
2. Navigate to the IOU details page and copy the URL
3. On other device paste the link and log in to account B

## Expected Result:
User B should navigate to IOU details

## Actual Result:
Error page appears on flash when navigate to IOU via link

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/44e2c1e7-146c-4b55-864e-90d4d39c13f2

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@CortneyOfstad  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Error page shows briefly when navigate to IOU link before login.

### What is the root cause of that problem?
One of the conditions for the not found to show is that if the report loading is already complete (!reportMetadata?.isLoadingInitialReportActions) but the report doesn't have `reportID`.
https://github.com/Expensify/App/blob/8ef59e7e9ac4d48695b2a54a5423f1cef70cc7b6/src/pages/home/ReportScreen.tsx#L464-L466

But why the report still doesn't exist even after the OpenReport loading is complete?

That's because when we open the IOU report link while logged out, we trigger the OpenReport to check whether it's a public room or not, and when the loading is complete, the `reportMetadata?.isLoadingInitialReportActions` becomes `false`.
https://github.com/Expensify/App/blob/8ef59e7e9ac4d48695b2a54a5423f1cef70cc7b6/src/libs/actions/Report.ts#L2585-L2588

When we log in, the IOU report is opened and we already wait for the app loading to complete before showing the not found page.
https://github.com/Expensify/App/blob/8ef59e7e9ac4d48695b2a54a5423f1cef70cc7b6/src/pages/home/ReportScreen.tsx#L453-L462

But OpenApp doesn't return all reports the user has, such as the IOU report.

In short, the not found shows because the previous request as non-logged in user fails.

### What changes do you think we should make in order to solve the problem?

If the OpenReport fails as a non-logged-in user, clears the `reportMetada` and `report`.
```
if (isFromDeeplink) {
    failureData = [
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT_METADATA}${reportID}`,
            value: null,
        },
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
            value: null,
        },
    ];
}
```

**Comment by CortneyOfstad** [comment]
This was assigned after I went OoO, so reapplying the `bug` label

**Comment by RachCHopkins** [comment]
VIP VSB is currently on hold. Just checking on next steps.

**Comment by RachCHopkins** [comment]
I think we are closing these for now.
